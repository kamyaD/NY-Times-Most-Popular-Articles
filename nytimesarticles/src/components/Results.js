import React, {useEffect, useState}from 'react';
import './css/Styles.css';

const Results= ()=>{
    const [data, setData] = useState([]);
    const [loading, setLoading] = useState(false);

    useEffect(() => {
        setLoading(true);
        fetch('https://api.nytimes.com/svc/mostpopular/v2/viewed/1.json?api-key=wOvranaebS0fZYSYy6oPnPLhIKk4kALO')
          .then((res) => res.json())
          .then((data) => {
            console.log(data);
            setData(data.results);
          })
          .catch((err) => {
            console.log(err);
          })
          .finally(() => {
            setLoading(false);
          });
      }, []);
    
      if (loading) {
        return <p data-testid="content">Data is loading...</p>;
      }
    
      return (
        <div className='card'>
         <h2>NY Times Most Read</h2>
          {data.map((item) => (
            <div className='container'>
            <div key={item.id} style={{margin:'1rem'}}>{item.title}</div>
            <div key={item.id} style={{display:'inline-block',marginLeft:40, color:'grey'}}>{item.byline}</div>
            <div key={item.id} style={{display:'inline-block',marginLeft:40, color:'grey'}}>{item.published_date}</div>
            </div>
          ))}
        </div>
      );
    
};

export default Results;