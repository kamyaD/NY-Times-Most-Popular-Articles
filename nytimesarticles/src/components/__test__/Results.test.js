import React from 'react';
import ReactDom from 'react-dom';
import Results from '../Results';

import { render} from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";

import renderer from 'react-test-renderer'; 

it("renders without crashing",()=>{
     const div= document.createElement("div");
     ReactDom.render(<Results/>, div)
})

it("renders Results correctly", ()=>{
  const {getByTestId}=render(<Results/>) 
  expect(getByTestId('content')).toHaveTextContent("Data is loading...")
})

it("matches snapshot", ()=>{
   const tree = renderer.create(<Results/>).toJSON(); 
   expect(tree).toMatchSnapshot();
})
