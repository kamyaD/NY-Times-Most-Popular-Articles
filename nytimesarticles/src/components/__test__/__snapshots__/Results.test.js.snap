// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="card"
>
  <h2>
    NY Times Most Read
  </h2>
</div>
`;
